packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1081,21 +1099,54 @@ export class DatabaseCluster extends DatabaseClusterNew {
   constructor(scope: Construct, id: string, props: DatabaseClusterProps) {
     super(scope, id, props);
 
-    const credentials = renderCredentials(this, props.engine, props.credentials);
-    const secret = credentials.secret;
+    if (
+      props.manageMasterUserPassword &&
+      (
+        props.credentials?.excludeCharacters ||
+        props.credentials?.password ||
+        props.credentials?.replicaRegions ||
+        props.credentials?.secret ||
+        props.credentials?.secretName ||
+        props.credentials?.usernameAsString
+      )
+    ) {
+      throw new Error('Only the `username` and `encryptionKey` credentials properties may be used when `manageMasterUserPassword` is true');
+    }
 
-    const cluster = new CfnDBCluster(this, 'Resource', {
-      ...this.newCfnProps,
-      // Admin
-      masterUsername: credentials.username,
-      masterUserPassword: credentials.password?.unsafeUnwrap(),
-    });
+    let cluster: CfnDBCluster;
+    if (!props.manageMasterUserPassword) {
```

Comment:
Can we remove the `if` statement given the above validation?
```ts
      const credentials = renderCredentials(this, props.engine, props.credentials);
      const secret = credentials.secret;

      const cluster = new CfnDBCluster(this, 'Resource', {
        ...this.newCfnProps,
        // Admin
        masterUsername: credentials.username,
        masterUserPassword: credentials.password?.unsafeUnwrap(),
        // Define kms key the RDS Managed Secret will use.
        ...props.manageMasterUserPassword && credentials.encryptionKey && {
          masterUserSecret: {
            kmsKeyId: credentials.encryptionKey.keyId,
          },
        },
      });

      this.clusterIdentifier = cluster.ref;
      this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;

      if (secret) {
        this.secret = secret.attach(this);
      }

Reply:
We cannot unfortunately. The only way renderCredentials doesn't create a new secret is if the credentials prop is passed the output from a `Credentials.[method]` call with a SecretsManager Secret object, or a password that's an instance of SecretValue. Since we disallow password and secret from being passed as props to credentials, the renderCredential function would always create a new secret. Not desirable.

Reply:
I see, thanks for clarifying 👍 
We could remove a bit of duplication by moving
```ts
      this.clusterIdentifier = cluster.ref;
      this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
```
after the `if` statement block.
Also, note the other comment on unit tests.

Reply:
Good catch, I'll get that changed, thanks!

Reply:
> I see, thanks for clarifying 👍 We could remove a bit of duplication by moving
> 
> ```ts
>       this.clusterIdentifier = cluster.ref;
>       this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
> ```
> 
> after the `if` statement block. Also, note the other comment on unit tests.

I just tested this change, and it can't be done. A secret is created in one of the `if` blocks, and the secret needs those values to be set, so the lines must be duplicated no matter what.